### ISSUE TITLE
[Bug]: Cannot share a volume between services in docker-compose

### ISSUE THREAD

**Issue by Signum** [coollabsio/coolify#3126]
### Description

I would like to have a persistent volume that can be accessed by 2 services in a docker-compose/stack.

The idea is to add an SFTP service to the stack so that a web developer can access the DocumentRoot of another web service to put files there.

Unfortunately Coolify rejects that change: "Duplicate entry found. Please use a different name."

### Minimal Reproduction (if possible, example repository)

```yaml
services:
  backend:
    image: example/database
    volumes:
      - db-data:/etc/data

  backup:
    image: backup-service
    volumes:
      - db-data:/var/lib/backup/data

volumes:
  db-data:
```

### Version

v4.0.0-beta.323

### Cloud?

- [ ] Yes
- [X] No

**Comment by peaklabs-dev** [comment]
@Signum This should be working, as I can not reproduce the error with this compose file:
```
services:
  backend:
    image: 'redis:alpine'
    volumes:
      - 'db-data:/etc/data'
  backup:
    image: 'redis:alpine'
    volumes:
      - 'db-data:/var/lib/backup/data'
volumes:
  db-data: null
  ```

**Comment by Signum** [comment]
It took me a bit to reproduce it. But now I managed to. Try this:

Create a new stack from the app store for "Wordpress with MariaDB". That will properly deploy and lead to this stack:

```yaml
services:
  wordpress:
    image: 'wordpress:latest'
    volumes:
      - 'wordpress-files:/var/www/html'
    environment:
      - SERVICE_FQDN_WORDPRESS
      - WORDPRESS_DB_HOST=mariadb
      - WORDPRESS_DB_USER=$SERVICE_USER_WORDPRESS
      - WORDPRESS_DB_PASSWORD=$SERVICE_PASSWORD_WORDPRESS
      - WORDPRESS_DB_NAME=wordpress
    depends_on:
      - mariadb
    healthcheck:
      test:
        - CMD
        - curl
        - '-f'
        - 'http://127.0.0.1'
      interval: 2s
      timeout: 10s
      retries: 10
  mariadb:
    image: 'mariadb:11'
    volumes:
      - 'mariadb-data:/var/lib/mysql'
    environment:
      - MYSQL_ROOT_PASSWORD=$SERVICE_PASSWORD_ROOT
      - MYSQL_DATABASE=wordpress
      - MYSQL_USER=$SERVICE_USER_WORDPRESS
      - MYSQL_PASSWORD=$SERVICE_PASSWORD_WORDPRESS
    healthcheck:
      test:
        - CMD
        - healthcheck.sh
        - '--connect'
        - '--innodb_initialized'
      interval: 5s
      timeout: 20s
      retries: 10
```

Everything works.

Now I am clicking "Edit Compose File" and am adding an sftp-server to that stack and am deploying…

```yaml
  sftp-server:
      image: emberstack/sftp
      environment:
        - USERS=demo:demo:1001:1001
      expose:
        - "2200"
      volumes:
        - wordpress-files:/home/demo/sftp
      ports:
        - "2200:2200"
```

That also works.

Then I realize that I want to mount the "wordpress-files" volume not at /home/demo/sftp but rather at /home/demo/web. So I'm changing the "sftp-server" service (actually only the "volumes:" definition) to:


```yaml
  sftp-server:
      image: emberstack/sftp
      environment:
        - USERS=demo:demo:1001:1001
      expose:
        - "2200"
      volumes:
        - wordpress-files:/home/demo/web
      ports:
        - "2200:2200"
```

I click on "Save" and am reliably getting the "Duplicate entry found. Please use a different name." error message. However when I change the path back to "/home/demo/sftp", I can save.

So basically sharing a volume works the first time. But if you decide to change the path where it gets mounted into a further service, you get this error.

Could you please try this?

**Comment by djsisson** [comment]
@Signum out of interest  if you remove the volume from that part of the compose file, save then readd with the correct place does that work?

**Comment by Signum** [comment]
1. Removed it.
1. Saved (worked).
1. Re-deployed the stack.
1. Added it again.
1. Saved (Duplicate entry found. Please use a different name)
1. Ignored that and re-deployed.
1. 500 Server Error:

Error: SQLSTATE[23505]: Unique violation: 7 ERROR: duplicate key value violates unique constraint "local_persistent_volumes_name_resource_id_resource_type_unique" DETAIL: Key (name, resource_id, resource_type)=(qggkswc4s048w0cs8wc0s004_wordpress-files, 29, App\Models\ServiceApplication) already exists. (Connection: pgsql, SQL: insert into "local_persistent_volumes" ("mount_path", "resource_id", "resource_type", "name", "updated_at", "created_at") values (/newpath, 29, App\Models\ServiceApplication, qggkswc4s048w0cs8wc0s004_wordpress-files, 2024-08-20 19:14:44, 2024-08-20 19:14:44) returning "id")

**Comment by b3n3w** [comment]
Having the same issue here. Need to share a volume over multiple containers.

**Comment by balloman** [comment]
Is there a fix or a version to revert to? Docker Compose is now just broken for me??

**Comment by b3n3w** [comment]
> Is there a fix or a version to revert to? Docker Compose is now just broken for me?? Is there an SOP for how to at least login and update the Postgres DB manually? I'm stuck with my entire project being undeployable until there is a fix

You can delete the missconfigured appplication env bei calling the API over
` https://<your-instance>/api/v1/projects/<project-id>/<environment-name>`. Just provide a Bearer token as Auth in the Header. You can create it over the UI `/security/private-key`

**Comment by ShadowArcanist** [comment]
This issue has been inactive for over a year. Closing this for now. Please comment if this is still an issue.
